source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/AvailableChargeReceiptData/AvailableChargeReceiptData.cs

```diff
@@ -45,6 +44,17 @@ public class AvailableChargeReceiptData : AvailableDataBase
             _reasonCodes = reasonCodes;
         }
 
+        /// <summary>
+        /// Used implicitly by persistence.
+        /// </summary>
+        // ReSharper disable once UnusedMember.Local
+        private AvailableChargeReceiptData(string recipientId, string originalOperationId)
+            : base(recipientId)
+        {
+            OriginalOperationId = originalOperationId;
+            _reasonCodes = new List<AvailableChargeReceiptDataReasonCode>();
+        }
```

Comment:
Wouldn't it be slightly better to refactor the existing code by initalizing the ReasonCode as follows:
```suggestion
            ReasonCodes = new ReadOnlyCollection<AvailableChargeReceiptDataReasonCode>(_reasonCodes);
        }
```
and make the property a get-only. This way we can avoid creating a read-only collection multiple times.